APMrover2/sailboat.cpp

```diff
@@ -207,11 +208,23 @@ void Sailboat::get_throttle_and_mainsail_out(float desired_speed, float &throttl
         throttle_out = 0.0f;
     }
 
+    // if we are motoring relax sails
+    if (motor_state == UseMotor::USE_MOTOR_ALWAYS) {
+        mainsail_out = 100.0f;
+        wingsail_out = 0.0f;
+        return;
+    }
+
+    // use PID controller to sheet out
+    float pid_offset = rover.g2.attitude_control.get_sail_out_from_heel(radians(sail_heel_angle_max), rover.G_Dt) * 100.0f;
+    pid_offset = constrain_float(pid_offset, 0.0f, 100.0f);
+
     //
     // mainsail control
     //
-    // if we are motoring or attempting to reverse relax the sail
-    if (motor_state == UseMotor::USE_MOTOR_ALWAYS || !is_positive(desired_speed)) {
```

Comment:
Is the removal of the motor_state and then lower down the pid_offset intentional?  These two changes seem unrelated..

Reply:
They have been moved up,  motor_state  is on line 212 and the PID offset has been moved so we can use it for both the mainsail and wingsail. The reason for this re-jig is that we can go backwards with a wing sail (no promising it will work well, but it is possible) but not a 'normal' sail.  Thanks for taking a look